cmd/create-index-snapshot/main.go

```diff
@@ -64,7 +64,7 @@ func run() int {
 	log = log.Level(level)
 
 	// Open the index database.
-	db, err := badger.Open(dps.DefaultOptions(flagIndex))
+	db, err := badger.Open(dps.DefaultOptions(flagIndex).WithReadOnly(true))
```

Comment:
When a writer opens the database, he will claim the lock. In order for a reader to then open it, he not only needs to open it with `WithReadOnly`, but _also_ with `WithBypassLockGuard`. So every reader needs to use both options, while writers should use neither.